templates/core/terraform/main.tf

```diff
@@ -14,7 +14,7 @@ provider "azurerm" {
   features {
     key_vault {
       purge_soft_delete_on_destroy    = var.keyvault_purge_protection_enabled ? false : true
-      recover_soft_deleted_key_vaults = true
+      recover_soft_deleted_key_vaults = false
```

Comment:
Why are we changing this? I believe it was set to true so could redeploy with the same tre_id?

Reply:
So I copied it from core which is actually false, and then had problems so added it to true in a previous PR and then am aligning it to core. What do you think?

Reply:
I actually think we should have it to true - have you had better luck with it set to true?

Reply:
IIRC, this used to be true and @damoodamoo set to `false` when looking into build issues.
I think that the issue with recovery is that when you recover a KeyVault, secrets etc also get brought back with it. So then Terraform fails because secrets already exist and need importing. So having as `false` and failing early ended up a better experience.

I _think_ that the underlying issues here were around the PR build destroy step failing to purge the KV, but I think that's all working now?

Reply:
Both set to false